src/util/movinginterquartilemean.cpp

```diff
@@ -0,0 +1,88 @@
+#include "movinginterquartilemean.h"
+
+MovingInterquartileMean::MovingInterquartileMean(const unsigned int listMaxSize)
+    : m_iListMaxSize(listMaxSize),
+      m_bChanged(true) {
+}
+
+MovingInterquartileMean::~MovingInterquartileMean() {};
+
+double MovingInterquartileMean::insert(double value) {
+    m_bChanged = true;
+
+    // Insert new value
+    if (m_list.isEmpty()) {
```

Comment:
you may want to use a QMap. It already stores the keys in order for you. You just need to insert a garbage value. You have later easy access to the sorted keys in a a QList.